ogre/src/OgreLidarVisual.cc

```diff
@@ -275,41 +342,105 @@ void OgreLidarVisual::Update()
       ignition::math::Vector3d noHitPt =
                   (axis * noHitRange) + this->offset.Pos();
 
-      // Draw the lines and strips that represent each simulated ray
-      if (i >= this->dataPtr->rayLines[j]->PointCount()/2)
-      {
-        this->dataPtr->rayLines[j]->AddPoint(startPt);
-        this->dataPtr->rayLines[j]->AddPoint(inf ? noHitPt : pt);
-
-        this->dataPtr->rayStrips[j]->AddPoint(startPt);
-        this->dataPtr->rayStrips[j]->AddPoint(inf ? startPt : pt);
+      // Update the lines and strips that represent each simulated ray.
 
-        this->dataPtr->noHitRayStrips[j]->AddPoint(startPt);
-        this->dataPtr->noHitRayStrips[j]->AddPoint(inf ? noHitPt : pt);
-      }
-      else
+      // For RAY_LINES Lidar Visual to be displayed
+      if (this->dataPtr->lidarVisType ==
+                LidarVisualType::LVT_RAY_LINES)
       {
-        this->dataPtr->rayLines[j]->SetPoint(i*2, startPt);
-        this->dataPtr->rayLines[j]->SetPoint(i*2+1, inf ? noHitPt : pt);
+        if (i >= this->dataPtr->rayLines[j]->PointCount()/2)
+        {
+          if (this->displayNonHitting || !inf)
+          {
+            this->dataPtr->rayLines[j]->AddPoint(startPt);
+            this->dataPtr->rayLines[j]->AddPoint(inf ? noHitPt : pt);
+          }
+        }
+        else
+        {
+          if (this->displayNonHitting || !inf)
+          {
+            this->dataPtr->rayLines[j]->SetPoint(i*2, startPt);
+            this->dataPtr->rayLines[j]->SetPoint(i*2+1, inf ? noHitPt : pt);
+          }
+        }
+      }
 
-        this->dataPtr->rayStrips[j]->SetPoint(i*2, startPt);
-        this->dataPtr->rayStrips[j]->SetPoint(i*2+1, inf ? startPt : pt);
+      // For TRIANGLE_STRIPS Lidar Visual to be displayed
+      else if (this->dataPtr->lidarVisType ==
+                LidarVisualType::LVT_TRIANGLE_STRIPS)
+      {
+        if (i >= this->dataPtr->rayLines[j]->PointCount()/2)
+        {
+          {
+            this->dataPtr->rayLines[j]->AddPoint(startPt);
+            this->dataPtr->rayLines[j]->AddPoint(inf ? noHitPt : pt);
+
+            this->dataPtr->rayStrips[j]->AddPoint(startPt);
+            this->dataPtr->rayStrips[j]->AddPoint(inf ? startPt : pt);
+
+            this->dataPtr->noHitRayStrips[j]->AddPoint(startPt);
+            this->dataPtr->noHitRayStrips[j]->AddPoint(inf ? noHitPt : pt);
+          }
+        }
+        else
+        {
+          {
+            this->dataPtr->rayLines[j]->SetPoint(i*2, startPt);
+            this->dataPtr->rayLines[j]->SetPoint(i*2+1, inf ? noHitPt : pt);
+
+            this->dataPtr->rayStrips[j]->SetPoint(i*2, startPt);
+            this->dataPtr->rayStrips[j]->SetPoint(i*2+1, inf ? startPt : pt);
+
+            this->dataPtr->noHitRayStrips[j]->SetPoint(i*2, startPt);
```

Comment:
when `displayNonHitting` is set to `false`, could we just to draw the `noHitRayStrips`?

Reply:
In this case, not drawing noHitRay Strips, will not help as the rayStrips will form a large triangle across the multiple hitting points, covering the middle area. I'll rewrite this part to avoid this behaviour..